media-libs/libsdl3/libsdl3-3.2.4.recipe

```diff
@@ -0,0 +1,74 @@
+SUMMARY="Simple Direct Media Layer 3.0"
+DESCRIPTION="Simple DirectMedia Layer is a cross-platform development library \
+designed to provide low level access to audio, keyboard, mouse, joystick, and \
+graphics hardware via OpenGL and Direct3D. It is used by video playback \
+software, emulators, and popular games."
+HOMEPAGE="https://www.libsdl.org/"
+COPYRIGHT="1997-2024 Sam Lantinga"
+LICENSE="Zlib"
+REVISION="1"
+SOURCE_URI="https://www.libsdl.org/release/SDL3-$portVersion.tar.gz"
+CHECKSUM_SHA256="2938328317301dfbe30176d79c251733aa5e7ec5c436c800b99ed4da7adcb0f0"
+SOURCE_DIR="SDL3-$portVersion"
+# PATCHES=""
+
+ARCHITECTURES="all"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+    libsdl3$secondaryArchSuffix = $portVersion compat >= 3.0
+    lib:libsdl3$secondaryArchSuffix = 0.200.4 compat >= 0
+    "
+REQUIRES="
+    haiku$secondaryArchSuffix
+    lib:libGL$secondaryArchSuffix
+    # lib:libglu$secondaryArchSuffix
+    "
+
+PROVIDES_devel="
+    libsdl3${secondaryArchSuffix}_devel = $portVersion
+    devel:libsdl3$secondaryArchSuffix = 0.200.10 compat >= 0
+    "
+REQUIRES_devel="
+    libsdl3$secondaryArchSuffix == $portVersion base
+    "
+
+BUILD_REQUIRES="
+    haiku${secondaryArchSuffix}_devel
+    devel:libgl$secondaryArchSuffix
+    # devel:libglu$secondaryArchSuffix
+    "
+BUILD_PREREQUIRES="
+    cmd:cmake
+    cmd:gcc$secondaryArchSuffix
+    cmd:ld$secondaryArchSuffix
+    cmd:ninja
+    cmd:pkg_config$secondaryArchSuffix
+    "
+
+BUILD()
+{
+    cmake . -Bbuild -GNinja $cmakeDirArgs \
+        -DSDL_STATIC=OFF \
+        -DCMAKE_BUILD_TYPE=Release
+    ninja -C build
+}
+
+INSTALL()
+{
+    ninja -C build install
+
+    rm $libDir/libSDL3_test.a
+    rm $libDir/cmake/SDL3/SDL3testTargets-release.cmake
```

Comment:
Any reason to remove this file?

Reply:
Just saw that building sdl3_image fails to find the static library, this shouldn't be removed I think and should be moved to the _devel package, then sed for the path with:
```
	sed -i 's,${_IMPORT_PREFIX}/lib',$developDir/lib, \
		$libDir/cmake/SDL3/SDL3testTargets-release.cmake
```

Reply:
Oh, I see. I saw that the sdl2 port was doing that so just followed that thinking there is some issues it on haiku,

Reply:
Normaly we delete the static libraries (you will get a policy error anyway when libfoo.so and libfoo.a is installed with haikuporter), in this case it's not about libfoo.a but libfoo_test.a and we probably should keep this.

Reply:
> ```
> 	sed -i 's,${_IMPORT_PREFIX}/lib',$developDir/lib, \
> 		$libDir/cmake/SDL3/SDL3testTargets-release.cmake
> ```

Does the original code already handle the secondaryArch subdir or should that be `$developLibDir` instead of `$developDir/lib`?

Reply:
My bad on this one :)

Reply:
Done